src/ios/BrotherPrinter.m

```diff
@@ -188,69 +144,130 @@ -(void)findNetworkPrinters:(CDVInvokedUrlCommand*)command {
+
+        _bluetoothManager = [BRPtouchBluetoothManager sharedManager];
+        [_bluetoothManager registerForBRDeviceNotifications];
+    }
+}
+
+-(void)_accessoryDidDisconnect:(NSNotification *)notification
+{
+    BRPtouchDeviceInfo *disconnectedAccessory = [[notification userInfo] objectForKey:BRDeviceKey];
+    NSLog(@"DisconnectedDevice:[%@]",[disconnectedAccessory description]);
+}
+
+-(void)_accessoryDidConnect:(NSNotification *)notification
+{
+    BRPtouchDeviceInfo *connectedAccessory = [[notification userInfo] objectForKey:BRDeviceKey];
+    NSLog(@"ConnectedDevice:[%@]",[connectedAccessory description]);
+}
+
+-(void)pairBluetoothPrinters:(CDVInvokedUrlCommand*)command {
+    [self checkBluetoothManager];
+
+    NSPredicate *pred = [NSPredicate predicateWithBlock:^ BOOL (NSString *evaluatedObject, NSDictionary * bindings) {
+        if (!evaluatedObject || ![evaluatedObject isKindOfClass: [NSString class]]) {
+            return NO;
+        }
+
+        __block BOOL keep = NO;
+        [[self suffixedPrinterList:supportedPrinterList] enumerateObjectsUsingBlock:^(NSString *printerName, NSUInteger index, BOOL *stop) {
+            keep = keep || [evaluatedObject hasPrefix:printerName];
+        }];
+
+        return keep;
+    }];
+
+    [_bluetoothManager brShowBluetoothAccessoryPickerWithNameFilter:pred];
+    [self.commandDelegate
+        sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK]
+              callbackId:command.callbackId];
 }
 
 -(void)findBluetoothPrinters:(CDVInvokedUrlCommand*)command {
-//	[self.commandDelegate runInBackground:^{
-        [self pairedDevicesWithCompletion:^(NSArray* bluetoothPrinters, NSError *error) {
-            if (error) {
+    [self.commandDelegate runInBackground:^{
+        [self checkBluetoothManager];
+        [self pairedDevicesWithCompletion:^(NSArray* pairedDevices, NSError* error) {
+            if (error != nil) {
                 [self.commandDelegate
-                    sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:[error localizedDescription]]
+                    sendPluginResult:[self errorResult:@"findBluetoothPrinters" withCode:1 withMessage:[NSString stringWithFormat:@"unable to find network printers: %@", [error localizedDescription]]]
                           callbackId:command.callbackId];
+
                 return;
             }
 
             [self.commandDelegate
-                sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:bluetoothPrinters]
-                      callbackId:command.callbackId];
+                 sendPluginResult:[CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:pairedDevices]
+                       callbackId:command.callbackId];
         }];
-//    }];
+     }];
 }
 
 -(void)findPrinters:(CDVInvokedUrlCommand*)command {
+//    [self.commandDelegate runInBackground:^{
```

Comment:
did this need to run in the background? did you put in foreground for testing? or does it not work in background?

Reply:
The function is a combination of searching for `findNetPrinters` and `findBluetoothPrinters`, the `findNetPrinters` portion cannot run in a background thread, but the `findBluetoothPrinters` can.